### PR TITLE
Ports SCP-173, 999, and 294

### PR DESCRIPTION
**Pull request by ro5490**
## About The Pull Request
Adds SCP-173, SCP-999, and SCP-294 plus their relevent effects, sounds, icons and so on.
Admin spawn only currently.

Original creators - JakeDaBoss and Kachnov / https://github.com/SS13-SCP13/SS13-SCP13
Hippie Port by - Toxici11i / https://github.com/HippieStation/HippieStation/pull/12224
Code adjustments by - Ro5490

## Why It's Good For The Game
Spooky SCP memes.
SCP-999 is like a reverse slime, it can latch onto people to heal them!
SCP-294 is a vending machine with hidden powers...
And SCP-173 will literally snap your neck for blinking.

## Changelog
:cl: Toxici11i, JakeDaBoss, Kachnov, Ro5490
add: Added SCP-173 and associated files
add: Added SCP-294 and associated files
add: Added SCP-999 and associated files
tweak: tweaked statuecode to work with SCP-173
soundadd: added sounds for SCP-173
imageadd: added some icons and images for the SCPs
code: Added #includes for SCP stuff to beestation.dme
/:cl:

### CODE REVIEW

**Comment by Crossedfall** [comment]
This is a partial port, correct? If so, please make sure you credit the original creators and link to their prs.

**Comment by ro5490** [comment]
> This is a partial port, correct? If so, please make sure you credit the original creators and link to their prs.

Added <3

**Comment by Crossedfall** [comment]
Danke. The next time you want to port something, please let me know. I'd like to teach you how to properly cherry-pick commits so that you can ensure that they maintain their attributions towards the original author(s). Since this looks to be a fairly large PR, I wont make you redo everything this time; but, keep it in mind for the future. 😄

**Comment by ro5490** [comment]
> Danke. The next time you want to port something, please let me know. I'd like to teach you how to properly cherry-pick commits so that you can ensure that they maintain their attributions towards the original author(s). Since this looks to be a fairly large PR, I wont make you redo everything this time; but, keep it in mind for the future. 😄

Thank you <3 I'll let you know!

**Comment by fighterslam** [comment]
how will blinking affect gameplay

**Comment by zeskorion** [comment]
> how will blinking affect gameplay

looking at the code, it seems to be an effect only when 173 is in view

> > Danke. The next time you want to port something, please let me know. I'd like to teach you how to properly cherry-pick commits so that you can ensure that they maintain their attributions towards the original author(s). Since this looks to be a fairly large PR, I wont make you redo everything this time; but, keep it in mind for the future. 😄
> 
> Thank you <3 I'll let you know!

according to ike, this is pretty shitcode, and i'm inclined to agree after a bit more scrutiny. Couldn't 173 be a statue subtype with some unique procs, and 999 a slime subtype? even 294 could be branched from a debug chem dispenser

**Comment by ro5490** [comment]
Its a port of a port, of course its pretty shitcode, but it works and runs pretty smoothly on my test server. feel free to look through and try and optimise it but this is the best I can do myself currently.

**Comment by zeskorion** [comment]
In a recent chat, I was made aware you dont even know what these things do codewise. Gonna close this, and help with a pr in a better state

File: beestation.dme [hunk 1]
```diff
@@ -2826,6 +2826,9 @@
 #include "code\modules\ruins\spaceruin_code\whiteshipruin_box.dm"
 #include "code\modules\security_levels\keycard_authentication.dm"
 #include "code\modules\security_levels\security_levels.dm"
+#include "code\modules\SCPmodular\scps\euclid\SCP-173.dm"
+#include "code\modules\SCPmodular\scps\euclid\SCP-294.dm"
+#include "code\modules\SCPmodular\scps\safe\scp999.dm"
```
Comment (ike709) [inline beestation.dme:2831]: Bad structure, all the DM files should just be in a single `scp` module.

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+	maxHealth = 5000
+	health = 5000
+	move_force = MOVE_FORCE_NORMAL
+	move_resist = MOVE_FORCE_EXTREMELY_STRONG
+	pull_force = MOVE_FORCE_EXTREMELY_STRONG
+	response_help = "touches"
+	response_disarm = "pushes"
+	a_intent = INTENT_HARM
+//	possible_a_intents = list(INTENT_HARM)
```
Comment (ike709) [inline code/modules/SCPmodular/scps/euclid/SCP-173.dm:16]: Don't leave commented-out code.

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+	harm_intent_damage = 1
+	obj_damage = 500
+	melee_damage_lower = 5000
```
Comment (ike709) [inline code/modules/SCPmodular/scps/euclid/SCP-173.dm:19]: > instant-killing

That has some pretty serious balance concerns, even if it is accurate to the SCP.

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+	// Check for darkness
+	var/turf/T = get_turf(loc)
+	if(T && destination && T.lighting_object)
+		if(T.get_lumcount()<0.1 && destination.get_lumcount()<0.1) // No one can see us in the darkness, right?
+			return null
+		if(T == destination)
+			destination = null
+
+	// We aren't in darkness, loop for viewers.
+	var/list/check_list = list(src)
```
Comment (ike709) [inline code/modules/SCPmodular/scps/euclid/SCP-173.dm:68]: Christ.

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+				if(!M.eye_blind)
+					if(next_blinks[M] == null)
+						next_blinks[M] = world.time+rand(15 SECONDS, 60 SECONDS)
+					return M
+		for(var/obj/mecha/M in view(world.view + 1, check)) //assuming if you can see them they can see you
+			if(M.occupant && M.occupant.client)
+				if(!M.occupant.eye_blind)
+					if(next_blinks[M.occupant] == null)
+						next_blinks[M.occupant] = world.time+rand(15 SECONDS, 60 SECONDS)
+					return M.occupant
```
Comment (ike709) [inline code/modules/SCPmodular/scps/euclid/SCP-173.dm:88]: Christ.

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+		for(var/mob/living/M in viewers(world.view + 1, check) - src)
+			if(M.client && !M.has_unlimited_silicon_privilege)
+				if(!M.eye_blind)
+					return M
+		for(var/obj/mecha/M in view(world.view + 1, check)) //assuming if you can see them they can see you
+			if(M.occupant && M.occupant.client)
+				if(!M.occupant.eye_blind)
+					return M.occupant
+	return null
+*/
```
Comment (ike709) [inline code/modules/SCPmodular/scps/euclid/SCP-173.dm:108]: No commented out code.

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+
+
+/mob/living/simple_animal/hostile/scp_173/Move(turf/NewLoc)
+	if(can_be_seen(NewLoc))
+		if(client)
+			to_chat(src, "<span class='warning'>You cannot move, there are eyes on you!</span>")
+		return 0
+	return ..()
+
+/mob/living/simple_animal/hostile/scp_173/movement_delay()
```
Comment (ike709) [inline code/modules/SCPmodular/scps/euclid/SCP-173.dm:118]: Isn't there a proper var for move delay?

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+			next_blinks[A] = null
+			continue
+		if(world.time >= next_blinks[A])
+			var/mob/living/carbon/human/H = A
+			if(H.stat) // Sleeping or dead people can't blink!
+				next_blinks[A] = null
+				continue
+			H.visible_message("<span class='notice'>[H] blinks.</span>")
+			H.blind_eyes(2)
+			next_blinks[H] = 10+world.time+rand(15 SECONDS, 60 SECONDS)
```
Comment (ike709) [inline code/modules/SCPmodular/scps/euclid/SCP-173.dm:139]: Jesus Christ.

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+
+
+
+
+
+/mob/living/simple_animal/hostile/scp_173/sentience_act()
+	faction -= "neutral"
+
+
+/mob/living/simple_animal/hostile/scp_173/AttackingTarget(/*var/atom/A*/) // doesnt work?
```
Comment (ike709) [inline code/modules/SCPmodular/scps/euclid/SCP-173.dm:149]: What's with the comment here?

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+				playsound(loc, pick('sound/scpsounds/scp/spook/NeckSnap1.ogg', 'sound/scpsounds/scp/spook/NeckSnap3.ogg'), 50, 1)
+				H.death()
+			if(!ishuman(H))
+				visible_message("<span class='danger'>[src] crushes [H] with raw force!</span>")
+				playsound(loc, pick('sound/scpsounds/scp/spook/NeckSnap1.ogg', 'sound/scpsounds/scp/spook/NeckSnap3.ogg'), 50, 1)
+				H.death()
+			if(!isliving(target))
+				visible_message("<span class='danger'>[src] crushes [H] with raw force!</span>")
+				playsound(loc, pick('sound/scpsounds/scp/spook/NeckSnap1.ogg', 'sound/scpsounds/scp/spook/NeckSnap3.ogg'), 50, 1)
+				H.death()
```
Comment (ike709) [inline code/modules/SCPmodular/scps/euclid/SCP-173.dm:170]: All of this code is the same except for one of the visible messages. You also already have an isliving check.

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+			else
+				return FALSE // this is just for error catching
+		else
+			return ..()
+/*			to_chat(src, "<span class='warning'><I>Why would we waste our energy attacking [H]</I></span>")
```
Comment (ike709) [inline code/modules/SCPmodular/scps/euclid/SCP-173.dm:175]: Commented-out code.

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+			return //doesnt allow 173 to attack apcs
+*/
+
+/mob/living/simple_animal/hostile/scp_173/DestroyPathToTarget()
+	if(!can_be_seen(get_turf(loc)))
+		..()
+
+
+/*
+/mob/living/simple_animal/hostile/scp_173/UnarmedAttack(var/atom/A)
```
Comment (ike709) [inline code/modules/SCPmodular/scps/euclid/SCP-173.dm:185]: Commented out code.

File: beestation.dme (hunk 1 above)
Comment (ro5490) [reply beestation.dme:2831]: Will fix by the next PR

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 4 above)
Comment (ro5490) [reply code/modules/SCPmodular/scps/euclid/SCP-173.dm:68]: This is generic /Tg/ Statuecode, just in 173's file
https://github.com/BeeStation/BeeStation-Hornet/blob/1cf388a0f1ef9fe54af524932a8028bd4bafece7/code/modules/mob/living/simple_animal/hostile/statue.dm

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 5 above)
Comment (ro5490) [reply code/modules/SCPmodular/scps/euclid/SCP-173.dm:88]: This is generic /Tg/ Statuecode, just in 173's file
https://github.com/BeeStation/BeeStation-Hornet/blob/1cf388a0f1ef9fe54af524932a8028bd4bafece7/code/modules/mob/living/simple_animal/hostile/statue.dm

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 8 above)
Comment (ro5490) [reply code/modules/SCPmodular/scps/euclid/SCP-173.dm:139]: /spell/aoe_turf/blindness is broke
this was the workaround, people blink withing 23 tiles of 173
If theres a better way to do it please point me in the right direction

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 9 above)
Comment (ro5490) [reply code/modules/SCPmodular/scps/euclid/SCP-173.dm:149]: currently changed that in test-code, was trying to get AI working and forgot to uncomment

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 10 above)
Comment (ro5490) [reply code/modules/SCPmodular/scps/euclid/SCP-173.dm:170]: visible_message(pick("<span class='danger'>[src] snaps [H]'s neck!</span>" , "<span class='danger'>[src] crushes [H] with raw force!</span>"))
				playsound(loc, pick('sound/scpsounds/scp/spook/NeckSnap1.ogg', 'sound/scpsounds/scp/spook/NeckSnap3.ogg'), 50, 1)
				H.death()
			else
Possible fix? untested yet.

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 7 above)
Comment (ro5490) [reply code/modules/SCPmodular/scps/euclid/SCP-173.dm:118]: `move_to_delay =`
seems like the right one, Will test

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 3 above)
Comment (ro5490) [reply code/modules/SCPmodular/scps/euclid/SCP-173.dm:19]: this is balanced by being unable to move or interact while being seen, gives enough time for mutiple crew members to whail on it til its dead
being frozen on the spot when in view and (due to blindess spell being broken) not being able to control when you can move or attack, plus having to wait it out anywhere from 25 to 70 seconds to be able to do something is a big nerf, gives someone enough time to run or call for help, and enough time for mutiple crew members to come by and beat on it

Meledamage is irrelevent here as the attacks call for H.death()

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 6 above)
Comment (ro5490) [reply code/modules/SCPmodular/scps/euclid/SCP-173.dm:108]: fixed in testing, next PR wont have this

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 11 above)
Comment (ro5490) [reply code/modules/SCPmodular/scps/euclid/SCP-173.dm:175]: fixed in testing, next PR wont have this commented out

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 12 above)
Comment (ro5490) [reply code/modules/SCPmodular/scps/euclid/SCP-173.dm:185]: fixed in testing, next PR wont have this commented out

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 2 above)
Comment (ro5490) [reply code/modules/SCPmodular/scps/euclid/SCP-173.dm:16]: fixed in testing, next PR wont have this

File: code/modules/SCPmodular/scps/euclid/SCP-173.dm (hunk 7 above)
Comment (ro5490) [reply code/modules/SCPmodular/scps/euclid/SCP-173.dm:118]: Works for AI control, but not for Player Control, swapping back to the proc movement delay call

**Pull request closed without merging**